pkg/controller/mobiledeveloperconsole/mobiledeveloperconsole_controller.go

```diff
@@ -100,54 +123,195 @@ func (r *ReconcileMobileDeveloperConsole) Reconcile(request reconcile.Request) (
 		return reconcile.Result{}, err
 	}
 
-	// Define a new Pod object
-	pod := newPodForCR(instance)
+	if instance.Status.Phase == mdcv1alpha1.PhaseEmpty {
+		instance.Status.Phase = mdcv1alpha1.PhaseProvision
+		err = r.client.Status().Update(context.TODO(), instance)
+		if err != nil {
+			reqLogger.Error(err, "Failed to update MDC resource status phase", "MDC.Namespace", instance.Namespace, "MDC.Name", instance.Name)
+			return reconcile.Result{}, err
+		}
+	} else if instance.Status.Phase == mdcv1alpha1.PhaseComplete {
+		reqLogger.Info("MDC resource is already in Complete phase. Doing nothing.", "MDC.Namespace", instance.Namespace, "MDC.Name", instance.Name)
```

Comment:
I mentioned this in my UPS PR [here](https://github.com/aerogear/unifiedpush-operator/pull/32/files#r298152789). I think it's incorrect to return optimistically like this here in the Reconcile method.

Think about the very next section of code where the ServiceAccount is created: once we get to PhaseComplete in the status, someone could accidentally delete the ServiceAccount (or any of the other resources), and this operator wouldn't fix that (even though it watches changes to those resources and gets triggered on changes for those).

Reply:
I think you're right.

Going to give this a try

Reply:
Yes, you're absolutely right...
Fixing this.